ds/shoelace/src/vue/x-alert.vue

```diff
@@ -17,49 +17,62 @@ License along with this software; if not, write to the Free
 Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 -->
-<script lang="ts" setup>
+<script setup lang="ts">
 import "@shoelace-style/shoelace/dist/components/alert/alert";
-import { computed, type PropType } from "vue";
+import "@shoelace-style/shoelace/dist/components/icon/icon";
+import type { AlertProps } from "@xwiki/cristal-dsapi";
+import { CIcon } from "@xwiki/cristal-icons";
+import { computed } from "vue";
 
-type Types = "" | "success" | "warning" | "error" | "info";
-
-const { type } = defineProps({
-  title: { type: String, required: true },
-  type: {
-    type: String as PropType<Types>,
-    required: true,
-  },
-  description: { type: String, required: true },
-});
+const { type } = defineProps<AlertProps>();
 
 const variant = computed(() => {
```

Comment:
By using [destructuring assignments](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) you can avoid the intermediate `variant` constant and avoid the odd `variant.variant` and `variant.icon` in the template.
```suggestion
const {variant, icon} = computed(() => {
```

Reply:
I tried but TypeScript complains when trying to destructure ComputedRefs

Reply:
ok, there is maybe something to be done with `toRef` but not sure if it would improve readability.
Let's ignore that point for now.